src/test/ui/closures/2229_closure_analysis/preserve_field_drop_order2.rs

```diff
@@ -0,0 +1,28 @@
+// run-pass
+// check-run-results
```

Comment:
I think we should add some test cases for paths that are multilevel  eg: a.b.c and a.b.d.